File: 1663. Smallest String With A Given Numeric Value/smallest-string-with-a-given-numeric-value.ts
// 1663. Smallest String With A Given Numeric Value (1/6/54192)
// Runtime: 153 ms (85.32%) Memory: 48.50 MB (86.24%) 

function getSmallestString(n: number, k: number): string {
    k -= n;
    const tail = "z".repeat(Math.floor(k / 25));
    const body = k % 25 === 0 ? "" : String.fromCharCode(k % 25 + 97);
    return (body + tail).padStart(n, "a");
};

File: 1663. Smallest String With A Given Numeric Value/smallest-string-with-a-given-numeric-value.c
// 1663. Smallest String With A Given Numeric Value (12/16/54191)
// Runtime: 37 ms (50.56%) Memory: 15.26 MB (75.66%) 

char* getSmallestString (int n, int k) {
    char* ans = (char*)calloc(n + 1, sizeof(char));

    for (int i = n - 1; i >= 0; i--) {
        int left = k - i;
        ans[i] = left >= 27 ? 'z' : 'a' + left - 1;
        k -= ans[i] - 'a' + 1;
    }

    return ans;
}
